pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -254,7 +312,11 @@ func (t *TopologyCache) getAllocations(numEndpoints int) map[string]Allocation {
 		minTotal += minimum
 		remainingMinEndpoints -= minimum
 		if remainingMinEndpoints < 0 {
-			return nil
+			return nil, &EventBuilder{
+				EventType: v1.EventTypeWarning,
+				Reason:    "TopologyAwareHintsDisabled",
+				Message:   fmt.Sprintf("%s (%d endpoints, %d zones)", InsufficientNumberOfEndpoints, numEndpoints, len(t.cpuRatiosByZone)),
```

Comment:
What's happening here is confusing, but I think the easiest way to describe this is something like this: "Unable to allocate minimum required endpoints to each zone without exceeding overload threshold."

Reply:
I think a simple test case to trigger this would be one where there were 4 endpoints and 3 zones.

Reply:
Thanks for clarifying this! 
You mentioned here that the change I've made makes sense. (was it before or after this comment?) https://github.com/kubernetes/kubernetes/pull/113544#discussion_r1014532325

Based on what you describe here I better change the event to a similar description right?

Reply:
Yeah sorry, I meant that an event here makes sense, but then came back to suggest a slightly different description.